src/app.py

```diff
@@ -1,26 +1,51 @@
 import json
+import logging
 import os
+import schedule
+import time
 from article import Article
 from publication import Publication
 from pymongo import MongoClient, UpdateOne
 
+# Open publications json file
 with open("publications.json") as f:
     publications_json = json.load(f)["publications"]
 
-publications = []
-articles = []
-for publication in publications_json:
-    pub = Publication(publication)
-    for entry in pub.get_feed():
-        articles.append(Article(entry, publication['slug']).serialize())
-    publications.append(pub.serialize())
+MONGO_ADDRESS = os.getenv("MONGO_ADDRESS")
 
-publication_upserts = [ UpdateOne({'_id':p['_id']}, {'$setOnInsert':p}, upsert=True) for p in publications ]
-article_upserts = [ UpdateOne({'articleUrl':a['articleUrl']}, {'$setOnInsert':a}, upsert=True) for a in articles ]
+# Get serialized publications
+publications = [Publication(p).serialize() for p in publications_json]
+publication_upserts = [
+    UpdateOne({"_id": p["_id"]}, {"$setOnInsert": p}, upsert=True) for p in publications
+]
+# Add publications to db
+with MongoClient(MONGO_ADDRESS) as client:
+    db = client.microphone
+    result = db.publications.bulk_write(publication_upserts)
 
-MONGO_ADDRESS = os.getenv("MONGO_ADDRESS")
-client = MongoClient(MONGO_ADDRESS)
+# Fucntion for gathering articles for running with scheduler
+def gather_articles():
+    articles = []
+    for publication in publications_json:
+        p = Publication(publication)
+        for entry in p.get_feed():
+            articles.append(Article(entry, publication["slug"]).serialize())
+
+    article_upserts = [
```

Comment:
what is an upsert haha

Reply:
it's insert + update lol 
Insert if the object is a new object // update if object already exists

Reply:
LOOOLLL damnnnnnn how did i not know that 😂  thankuu